Assets/SEE/Controls/Actions/MarkAction.cs

```diff
@@ -0,0 +1,118 @@
+﻿using System.Collections.Generic;
+using SEE.GO;
+using SEE.Utils.History;
+using UnityEngine;
+using SEE.Game.SceneManipulation;
+using SEE.Utils;
+using SEE.Net;
+
+namespace SEE.Controls.Actions
+{
+    internal class MarkAction : AbstractPlayerAction
+    {
+        /// <summary>
+        /// The node that was marked when this action was executed. It is saved so
+        /// action can be removed on Undo().
+        /// </summary>
+        private GameObject markedNode;
+
+        /// <summary>
+        /// Memento capturing the data necessary to re-do this action.
+        /// </summary>
+        private Memento memento;
+
+        /// <summary>
+        /// The information we need to re-mark a node.
+        /// </summary>
+        private struct Memento
```

Comment:
If the memento remembers only one value, you could as well do without a Memento struct.